otel-extension/src/main/java/akka/actor/impl/ActorCellReceivedAdvice.java

```diff
@@ -0,0 +1,44 @@
+package akka.actor.impl;
+
+import akka.actor.ActorContext;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.instrumentation.api.field.VirtualField;
+import io.scalac.mesmer.core.util.Interval$;
+import io.scalac.mesmer.otelextension.instrumentations.akka.actor.Instruments;
+import io.scalac.mesmer.otelextension.instrumentations.akka.actor.InstrumentsProvider;
+import io.scalac.mesmer.otelextension.instrumentations.akka.actor.impl.otel.ActorCellInstrumentationState;
+import java.util.Objects;
+import net.bytebuddy.asm.Advice;
+
+public class ActorCellReceivedAdvice {
+
+  @Advice.OnMethodEnter
+  public static long enter(@Advice.This ActorContext self) {
+
+    return System.nanoTime();
+  }
+
+  @Advice.OnMethodExit(onThrowable = Throwable.class)
+  public static void exit(
+      @Advice.This ActorContext self,
+      @Advice.Thrown Throwable exception,
+      @Advice.Enter long started) {
+    long interval = Interval$.MODULE$.toMillis(System.nanoTime() - started);
+
+    Attributes attrs = VirtualField.find(ActorContext.class, Attributes.class).get(self);
+    Instruments instruments = InstrumentsProvider.instance();
+    ActorCellInstrumentationState state =
+        VirtualField.find(ActorContext.class, ActorCellInstrumentationState.class).get(self);
+
+    if (Objects.nonNull(attrs) && Objects.nonNull(state)) {
```

Comment:
This kind of check seems to be used quite often - could you create a function for it?

Reply:
I don't really see how this will make the code clearer - a couple of null checks state pretty well what is going on and and It's not not a lot of repetition

Reply:
It's not only about cleanliness - this check is repeated 6 times in multiple advice classes, so we could:
 - reuse it and never forgot about checking one of the parameters
 - have it documented in one place, even if the method name would be self-documenting (eg. checkPreconditions).

However, shouldn't we throw when situations like this occur? If for some reason we will have attrs or state null we will silently fail. Shouldn't Mesmer "die" in case something as terrible as this happens?